torch/csrc/jit/codegen/cuda/executor_utils.cpp

```diff
@@ -980,20 +981,55 @@ std::vector<IterDomain*> getParallelBindingsIterDomains(
   return parallel_ids;
 }
 
+void insertParallelExtent(
+    GpuLower& lower,
+    IterDomain* binding_id,
+    const std::unique_ptr<ParallelExtentMap>& parallel_iter_extents_ptr) {
+  auto kir_extent = lower.lowerValue(binding_id->extent());
+  const auto it =
+      parallel_iter_extents_ptr->find(binding_id->getParallelType());
+  if (it != parallel_iter_extents_ptr->end()) {
+    it->second.push_back(kir_extent);
+  } else {
+    parallel_iter_extents_ptr->operator[](binding_id->getParallelType()) = {
+        kir_extent};
+  }
+}
+
 std::unique_ptr<ParallelExtentMap> getParallelIterExtents(
     GpuLower& lower,
     std::vector<IterDomain*>& parallel_binding_ids) {
   auto parallel_iter_extents_ptr = std::make_unique<ParallelExtentMap>();
   for (auto id : parallel_binding_ids) {
-    // TODO(kir): we should rewrite this logic based on the Kernel object
-    auto kir_extent = lower.lowerValue(id->extent());
-    const auto it = parallel_iter_extents_ptr->find(id->getParallelType());
-    if (it != parallel_iter_extents_ptr->end()) {
-      it->second.push_back(kir_extent);
-    } else {
-      parallel_iter_extents_ptr->operator[](id->getParallelType()) = {
-          kir_extent};
+    insertParallelExtent(lower, id, parallel_iter_extents_ptr);
+  }
+
+  return parallel_iter_extents_ptr;
+}
+
+std::unique_ptr<ParallelExtentMap> getSimplifiedParallelIterExtents(
```

Comment:
Can `ParallelDimensionMap` be used here? It would completely eliminate the analysis as the map should have the extent information for all parallel types. It's currently only available during the lowering, but should be possible to keep it after lowering.

Reply:
I think so, and the analyses done are along the same line, and actually `ParallelDimensionMap` does more optimization than `getSimplifiedParallelIterExtents`, which could help more. I'd prefer doing the integration in a follow up though as this function will very likely be moved into the `PreComputedInteger` after #1086 is merged.